src/shared/user-record-show.c

```diff
@@ -585,6 +585,22 @@ void user_record_show(UserRecord *hr, bool show_full_group_info) {
 
         if (hr->service)
                 printf("     Service: %s\n", hr->service);
+
+        if (!strv_isempty(hr->self_modifiable_fields))
+                STRV_FOREACH(i, hr->self_modifiable_fields)
+                        printf(i == hr->self_modifiable_fields ?
+                               " Self Modify: %s\n" :
+                               "              %s\n", *i);
+        else if (!strv_isempty(hr->self_modifiable_blobs) || !strv_isempty(hr->self_modifiable_privileged))
+                printf(" Self Modify: %s(no fields)%s\n", ansi_grey(), ansi_normal());
```

Comment:
hmm, this "no fields" is a bit misleading, no?

After all user_record_self_modifiable_fields() will take effect in that case, and list a bunch of default fields.

Dunno what the best approach here is. Could be either to just dump what user_record_self_modifiable_fields() returns here (which might be annoyingly long, but then again very accurate). Or we could show "(default fields)" in case the field is NULL, and "(no fields)" in case the field is non-NULL but the array is empty?